extensions/caffeine/runtime/pom.xml

```diff
@@ -21,6 +21,10 @@
             <groupId>com.github.ben-manes.caffeine</groupId>
             <artifactId>caffeine</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.ben-manes.caffeine</groupId>
+            <artifactId>jcache</artifactId>
```

Comment:
This comes with a few [direct/transitive dependencies](https://mvnrepository.com/artifact/com.github.ben-manes.caffeine/jcache/3.1.1) that may not be needed for most projects. Will everything be removed at build time?

FYI Quarkus has its own caching API (`quarkus-cache`) that does purposely not implement JSR107 (see #3306 for the reasons behind that choice).

@galderz @Sanne Do you think it's okay to introduce JCache in `quarkus-caffeine`?

Reply:
Yes, I saw the `quarkus-cache` but the JSR107 is needed in order to support `camel-quarkus-jcache` extension.
Not sure about dependencies though, I know that javax.inject is not needed and it is excluded in the application pom.

Reply:
I would prefer to not integrate support for JCache; my thoughts can be summarized around two points:
 - unclear future, especially as we are migrating to Jakarta EE 10 soon; it's not a good time to add further complexity in this migration; https://github.com/jsr107/jsr107spec/issues/415#issuecomment-1016536436
 - and indeed we have a Caching API which we believe is much better :)

Tell me more about `camel-quarkus-jcache` and why is it essential to support it?

cc/ @karesti @ppalaga

Reply:
Thanks for explaining the background, @Sanne! Indeed it seems to make little sense to strain quarkus-caffeine with any JSR107 legacy. `camel-quarkus-jcache` does not rely on any JSR107 annotations, just on its SPI. I am in favor of solving this on Camel Quarkus side as sketched in https://github.com/apache/camel-quarkus/issues/4092

Reply:
cool. Should we close this then? All good for you @javaduke ?